main.py

```diff
@@ -62,331 +61,290 @@
 )
 from fastlane_bot.utils import find_latest_timestamped_folder
 from run_blockchain_terraformer import terraform_blockchain
-
+import argparse
 
 load_dotenv()
 
 
-@click.command()
-@click.option(
-    "--cache_latest_only",
-    default=True,
-    type=bool,
-    help="Set to True for production. Set to False for " "testing / debugging",
-)
-@click.option(
-    "--backdate_pools",
-    default=False,
-    type=bool,
-    help="Set to False for faster testing / debugging",
-)
-@click.option(
-    "--static_pool_data_filename",
-    default="static_pool_data",
-    help="Filename of the static pool data.",
-)
-@click.option(
-    "--arb_mode",
-    default="multi_pairwise_all",
-    help="See arb_mode in bot.py",
-    type=click.Choice(
-        [
+def process_arguments(args):
+    """
+    Process and transform command line arguments.
+
+    :param args: Namespace object containing command line arguments.
+    :return: Processed arguments.
+    """
+    # Define the transformations for each argument
+    transformations = {
+        "backdate_pools": lambda x: x == "True",
```

Comment:
Is this the standard way of doing it?
I assume that you want a configuration parameter which can be either mentioned or omitted, but with no additional input following that, so it serves strictly as a boolean flag.
Wouldn't it be more correct to use something like:
```
lambda x: True if x else False
```
Or even:
```
lambda x: bool(x)
```
Same goes for a few other similar ones below...?

Reply:
Your suggestions do not handle the context properly. The command-line arguments are received as strings. I have attached three example usages, the first two are your suggestions, and the third is the existing code. You'll notice that only the third result gives the correct output. 
<img width="480" alt="image" src="https://github.com/bancorprotocol/fastlane-bot/assets/46603283/f636bcfd-2d3a-4594-a373-331da9688b9e">

Reply:
That's not how I meant for these boolean flags to be used.

What I meant is if you specify `--someBooleanOption` (without any additional value ascribed to it), then it means that you want that option enabled. And if you do not specify`--someBooleanOption`, then it means that you do not want this option enabled.

For this type of usage, I believe that `lambda x: True if x else False` should do the job.

Reply:
Please test your proposed changes prior to suggesting their use. Argparse has certain behaviors which are out of our control and out of scope for this PR. I am not familiar with any way to achieve what you're suggesting here, so please propose the change you'd like to see, if indeed it is possible.

Reply:
Something like this, for example:
```
from argparse import ArgumentParser
from argparse import BooleanOptionalAction

parser = ArgumentParser()
parser.add_argument('--boolean_option', action=BooleanOptionalAction)

args = parser.parse_args()
print(args.boolean_option)
```
It returns `True` when `--boolean_option` is specified, and `None` when `--boolean_option` is omitted.

If the type-duality matters, then you can of course convert it to `bool`, which will give you `True` and `False`.

Reply:
OK this is now updated in a way similar to what you proposed.